flixel/input/actions/FlxActionManager.hx

```diff
@@ -0,0 +1,735 @@
+				{
+					index = register.steamControllerSets[id];
+				}
+			case FlxInputDevice.ALL:
+				switch(deviceID)
+				{
+					case FlxInputDeviceID.ALL: index = register.gamepadAllSet;
+				}
+			default: index = -1;
+		}
+		
+		if (index >= 0 && index < sets.length)
+		{
+			return sets[index];
+		}
+		
+		return null;
+	}
+	
+	#if steamwrap
+	/**
+	 * Load action sets from a steamwrap ControllerConfig object
+	 * @param	Config	ControllerConfig object derived from your game's "game_actions_XYZ.vdf" file
+	 * @param	CallbackDigital	Callback function for digital actions
+	 * @param	CallbackAnalog	Callback function for analog actions
+	 * @return	The number of new FlxActionSets created and added. 0 means nothing happened.
+	 */
+	public function initSteam(Config:ControllerConfig, CallbackDigital:FlxActionDigital->Void, CallbackAnalog:FlxActionAnalog->Void):Int
+	{
+		var i:Int = 0;
+		for (set in Config.actionSets)
+		{
+			if (addSet(FlxActionSet.fromSteam(set, CallbackDigital, CallbackAnalog)) != -1)
+			{
+				i++;
+			}
+		}
+		
+		onChange();
+		
+		return i;
+	}
+	#end
+	
+	/**
+	 * Load action sets from a parsed JSON object
+	 * @param	data	JSON object parsed from the same format that "exportToJSON()" outputs
+	 * @param	CallbackDigital	Callback function for digital actions
+	 * @param	CallbackAnalog	Callback function for analog actions
+	 * @return	The number of new FlxActionSets created and added. 0 means nothing happened.
+	 */
+	public function initFromJSON(data:Dynamic, CallbackDigital:FlxActionDigital->Void, CallbackAnalog:FlxActionAnalog->Void):Int
+	{
+		if (data == null) return 0;
+		
+		var i:Int = 0;
+		var actionSets:Array<Dynamic> = Reflect.hasField(data, "actionSets") ? Reflect.field(data, "actionSets") : null;
+		if (actionSets == null) return 0;
+		
+		for (set in actionSets)
+		{
+			if (addSet(FlxActionSet.fromJSON(set, CallbackDigital, CallbackAnalog)) != -1)
+			{
+				trace("added set : " + sets[sets.length - 1].name);
+				i++;
+			}
+		}
+		
+		onChange();
+		
+		return i;
+	}
+	
+	@:access(flixel.input.actions.FlxAction)
+	public function exportToJSON():String
+	{
+		var space:String = "\t";
+		return Json.stringify({"actionSets":sets}, function(key:Dynamic, value:Dynamic):Dynamic{
+			
+			if (Std.is(value, FlxAction))
+			{
+				var fa:FlxAction = cast value;
+				return fa.name;
+			}
+			if (Std.is(value, FlxActionSet))
+			{
+				var fas:FlxActionSet = cast value;
+				return {
+					"name": fas.name,
+					"digitalActions": fas.digitalActions,
+					"analogActions": fas.analogActions
+				}
+			}
+			
+			return value;
+			
+		}, space);
+	}
+	
+	/**
+	 * Remove a set from the manager
+	 * @param	Set	The set you want to remove
+	 * @param	Destroy	Whether to also destroy the set or just remove it from the list
+	 * @return	whether the action was found & successfully removed
+	 */
+	public function removeSet(Set:FlxActionSet, Destroy:Bool = true):Bool
+	{
+		var success = false;
+		for (i in 0...sets.length)
```

Comment:
Should actually use `contains()` instead of `indexOf()` from `FlxArrayUtil` via static extension.